src/virtual_lodf_calculations.jl

```diff
@@ -0,0 +1,282 @@
+"""
+Power Transfer Distribution Factors (PTDF) indicate the incremental change in
+real power that occurs on transmission lines due to real power injections
+changes at the buses.
+
+The PTDF struct is indexed using the Bus numbers and branch names.
+
+# Arguments
+- `K::KLU.KLUFactorization{Float64, Int}`:
+        LU factorization matrices of the ABA matrix, evaluated by means of KLU
+- `BA::SparseArrays.SparseMatrixCSC{Float64, Int}`:
+        BA matric
+- `ref_bus_positions::Set{Int}`:
+        Vector containing the indexes of the columns of the BA matrix corresponding
+        to the refence buses
+- `dist_slack::Vector{Float64}`:
+        Vector of weights to be used as distributed slack bus.
+        The distributed slack vector has to be the same length as the number of buses.
+- `axes<:NTuple{2, Dict}`:
+        Tuple containing two vectors (the first one showing the branches names,
+        the second showing the buses numbers).
+- `lookup<:NTuple{2, Dict}`:
+        Tuple containing two dictionaries, the first mapping the branches
+        and buses with their enumerated indexes.
+- `temp_data::Vector{Float64}`:
+        temporary vector for internal use.
+- `cache::RowCache`:
+        cache were PTDF rows are stored.
+- `subnetworks::Dict{Int, Set{Int}}`:
+        dictionary containing the subsets of buses defining the different subnetwork of the system.
+- `tol::Base.RefValue{Float64}`:
+        tolerance related to scarification and values to drop.
+"""
+struct VirtualLODF{Ax, L <: NTuple{2, Dict}} <: PowerNetworkMatrix{Float64}
+    K::KLU.KLUFactorization{Float64, Int}
+    BA::SparseArrays.SparseMatrixCSC{Float64, Int}
+    A::SparseArrays.SparseMatrixCSC{Int8, Int}
+    PTDF_A_diag::Vector{Float64}
+    ref_bus_positions::Set{Int}
+    dist_slack::Vector{Float64}
+    axes::Ax
+    lookup::L
+    valid_ix::Vector{Int}
+    temp_data::Vector{Float64}
+    cache::RowCache
+    subnetworks::Dict{Int, Set{Int}}
+    tol::Base.RefValue{Float64}
+end
+
+function _get_PTDF_A_diag(
```

Comment:
This code is duplicated. We probably can get away with making a single function for this